### PR TITLE
Added BeehiveTileEntity interface for easier extensibility of Beehives

### PR DESCRIPTION
**Pull request by JaisDK**

### CODE REVIEW

File: patches/minecraft/net/minecraft/tileentity/IBeehiveTilEntity.java.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--- a/net/minecraft/tileentity/IBeehiveTileEntity.java
++++ b/net/minecraft/tileentity/IBeehiveTileEntity.java
+@@ -0,0 +1,9 @@
++package net.minecraft.tileentity;
++
++import net.minecraft.entity.Entity;
++
++public interface IBeehiveTileEntity {
```
Comment (KnightMiner) [inline patches/minecraft/net/minecraft/tileentity/IBeehiveTilEntity.java.patch]: Forge does not add new interfaces to the `net.minecraft` package. Add it to a Forge package and implement it in the vanilla block

File: patches/minecraft/net/minecraft/entity/passive/BeeEntity.java.patch [hunk 2]
```diff
@@ -0,0 +1,53 @@
+--- a/net/minecraft/entity/passive/BeeEntity.java
++++ a/net/minecraft/entity/passive/BeeEntity.java
+@@ -68,7 +68,7 @@ import net.minecraft.state.properties.DoubleBlockHalf;
+ import net.minecraft.tags.BlockTags;
+ import net.minecraft.tags.ItemTags;
+ import net.minecraft.tags.Tag;
+-import net.minecraft.tileentity.BeehiveTileEntity;
++import net.minecraft.tileentity.IBeehiveTileEntity;
```
Comment (KnightMiner) [inline patches/minecraft/net/minecraft/entity/passive/BeeEntity.java.patch]: Do not change imports in patches. Use fully qualified names for new class references, and leave now unused imports alone.

**Comment by DaemonUmbra** [comment]
After talking in Discord, apparently the patches were generated with git diff rather than genPatches.
Walking them through it now.

File: patches/minecraft/net/minecraft/tileentity/IBeehiveTilEntity.java.patch (hunk 1 above)
Comment (JaisDK) [reply patches/minecraft/net/minecraft/tileentity/IBeehiveTilEntity.java.patch]: Where would be the best place to put this in Forge? net.minecraftforge.common.extensions.IForgeBeehiveTileEntity ?

**Pull request closed without merging**